tests/trainer/test_checkpoint.py

```diff
@@ -213,6 +213,88 @@ def test_load_weights(
     )
 
 
+@pytest.mark.timeout(90)
+@pytest.mark.parametrize("device_hparams", [
+    pytest.param(CPUDeviceHparams(), id="cpu"),
+    pytest.param(GPUDeviceHparams(), id="gpu", marks=pytest.mark.gpu),
+])
+@pytest.mark.parametrize("save_overwrite", [
+    True,
+    False,
+])
+def test_save_overwrite(
+    device_hparams: DeviceHparams,
+    composer_trainer_hparams: TrainerHparams,
+    save_overwrite: bool,
```

Comment:
Mega nit: Is it possible to wrap the `save_overide` parameterization into the existing checkpoint tests? This code looks to be slightly duplicated from what's already there?

I know this is a bit out of the scope of the PR, but the checkpointing tests are getting quite messy and hard to maintain given all of the different parameterizations. As part of the graceful resumption, I think it would be good to clean it up with better shared helpers and to de-yahpify the tests to reduce code duplication. OK if that happens in another PR or this one, but just mentioning this now as this function (as written) will add to the future cleanup.

Reply:
Ack, but will do after this https://github.com/mosaicml/composer/pull/962 in a separate PR